server/routerlicious/packages/services-utils/src/auth.ts

```diff
@@ -52,3 +52,32 @@ export function generateUser(): IUser {
 
     return randomUser;
 }
+
+/**
+ * Validates a JWT token to authorize routerlicious and returns decoded claims.
+ * An undefined return value indicates invalid claims.
+ */
+export function validateTokenClaims(
+    token: string,
+    documentId: string,
+    tenantId: string,
+    maxTokenLifetimeSec: number,
+    isTokenExpiryEnabled: boolean): ITokenClaims {
+    const claims = jwt.decode(token) as ITokenClaims;
+
+    if (!claims
+        || claims.documentId !== documentId
+        || claims.tenantId !== tenantId
+    ) {
+        return undefined;
+    }
+
+    if (isTokenExpiryEnabled === true) {
+        const now = Math.round((new Date()).getTime() / 1000);
+        if (now >= claims.exp || claims.exp - claims.iat > maxTokenLifetimeSec) {
+            return undefined;
+        }
+    }
+
+    return claims;
+}
```

Comment:
Copy the file from services-client and use jsonwebtoken instead of jwtdecode. I added more code there.